versioned_docs/version-latest/learn/monitoring.md

```diff
@@ -0,0 +1,723 @@
+      timeout: 5s
+      interval: 30s
+      retries: 60
+#This maps port 9615 on the host where Prometheus is running to the same port 9615 within Docker container.
+    expose:
+      - "9615:9615"
+  
+  farmer:
+    depends_on:
+      node:
+        condition: service_healthy
+    image: ghcr.io/subspace/farmer:gemini-3g-2023-dec-22
+    volumes:
+      - farmer-data:/var/subspace:rw
+    ports:
+      - "0.0.0.0:30533:30533/udp"
+      - "0.0.0.0:30533:30533/tcp"
+    restart: unless-stopped
+    command:
+      [
+        "farm",
+        "--node-rpc-url", "ws://node:9944",
+        "--listen-on", "/ip4/0.0.0.0/udp/30533/quic-v1",
+        "--listen-on", "/ip4/0.0.0.0/tcp/30533",
+        "--reward-address", "st7GBwATPqtBkK5y4uXbV52euszPpFPw7wmkF8FywEqJaf8uP",
+        "path=/var/subspace,size=100G"
+      ]
+volumes:
+  node-data:
+  farmer-data:            
+```
+> The above section should be copied from the [official Subspace Docker Compose
+guide](https://docs.subspace.network/docs/farming-&-staking/farming/advanced-cli/cli-install)
+    
+> The below section defines the containers we need for the monitoring stack
+    
+```bash
+    grafana:
+        image: grafana/grafana:latest
+        container_name: grafana
+        volumes:
+        - grafana_data:/var/lib/grafana
+        - ./grafana/provisioning:/etc/grafana/provisioning
+        environment:
+        - GF_SECURITY_ADMIN_USER=ADMIN
+        - GF_SECURITY_ADMIN_PASSWORD=5u65pac3
+        - GF_USERS_ALLOW_SIGN_UP=false
+        restart: unless-stopped
+        ports:
+        - 3000:3000
+
+    prometheus:
+        image: prom/prometheus:latest
+        container_name: prometheus
+        restart: unless-stopped
+    volumes:
+        - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml
+        - prometheus_data:/prometheus
+        command:
+        - '--config.file=/etc/prometheus/prometheus.yml'
+        - '--storage.tsdb.path=/prometheus'
+        - '--web.console.libraries=/etc/prometheus/console_libraries'
+        - '--web.console.templates=/etc/prometheus/consoles'
+        - '--web.enable-lifecycle'
+        expose:
+        - 9090:9090
+
+    node-exporter:
+        image: prom/node-exporter:latest
+        container_name: node-exporter
+        restart: unless-stopped
+        volumes:
+        - /proc:/host/proc:ro
+        - /sys:/host/sys:ro
+        - /:/rootfs:ro
+        command:
+        - '--path.procfs=/host/proc'
+        - '--path.rootfs=/rootfs'
+        - '--path.sysfs=/host/sys'
+        - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+        expose:
+        - 9100
+    # The above section defines the containers we need for the monitoring stack
+    volumes:
+    # These first two volumes are required by the default setup from the official guide
+    node-data:
+    farmer-data:
+    # These next volumes are required by the monitoring solution
+    grafana_data:
+    prometheus_data:
+```
+> End of the file
```

Comment:
This is not being displayed correctly, perhaps can be removed.

Reply:
changed 

> End of file

for
&#x2191; &#x2191; The above section defines the containers we need for the monitoring stack &#x2191;&#x2191;